configure.ac

```diff
@@ -4132,16 +4131,17 @@ AS_VAR_IF([with_system_libmpdec], [yes],
       ], [const char *x = mpd_version();])],
       [have_mpdec=yes],
       [have_mpdec=no])
-  ])],
-  [AC_MSG_WARN([m4_normalize([
-     the bundled copy of libmpdecimal is scheduled for removal in Python 3.15;
+  ])])
+AS_VAR_IF([with_system_libmpdec], [no],
+          [AC_MSG_WARN([m4_normalize([
+     the bundled copy of libmpdecimal is scheduled for removal in Python 3.16;
      consider using a system installed mpdecimal library.])])])
 
 AS_IF([test "$with_system_libmpdec" = "yes" && test "$have_mpdec" = "no"],
       [AC_MSG_WARN([m4_normalize([
-         no system libmpdecimal found; falling back to bundled libmpdecimal
-         (deprecated and scheduled for removal in Python 3.15)])])
-       USE_BUNDLED_LIBMPDEC()])
+         no system libmpdecimal found; falling back to pure-Python version
+         for the decimal module])])
+       AS_VAR_SET([py_cv_module_]_decimal, [n/a])])
```

Comment:
`AC_MSG_ERROR`? I think opting in to the pure Python version should be explicit.

Reply:
> AC_MSG_ERROR ?

Well, that could be an option.

Though, more complex wrt implementation: all linux jobs will fail, unless we either provide system libmpdec or change ./configure invocations to use a new option.

> I think opting in to the pure Python version should be explicit.

I'm not sure it's useful.  After all, the pure-Python version is always available as the ``_pydecimal.py``.

Did you suggest a new option like ``--with-purepython-decimal``?

Reply:
A warning is actually ok, but users may not see it and then, they would have a slow version of decimal. OTOH, an error might be too abrupt but at least the tansition would be more explicit and would force people to upgrade if they want an efficient way to do it.

I'm +0.25 for AC_MSG_ERROR just to force people to update. We can add an option `--allow-fallback-to-pydecimal` to handle CI possible failures, though this means that devs should be aware that their C code might not be tested in the CI due to that.

Reply:
A warning is actually ok, but users may not see it and then, they would have a slow version of decimal. OTOH, an error might be too abrupt but at least the tansition would be more explicit and would force people to upgrade if they want an efficient way to do it.

I'm +0.25 for AC_MSG_ERROR just to force people to update. We can add an option `--allow-fallback-to-pydecimal` to handle CI possible failures, though this means that devs should be aware that their C code might not be tested in the CI due to that.